Source/WebCore/dom/TrustedType.cpp

```diff
@@ -99,22 +99,22 @@ ASCIILiteral trustedTypeToCallbackName(TrustedType trustedType)
 // https://w3c.github.io/trusted-types/dist/spec/#process-value-with-a-default-policy-algorithm
 std::variant<std::monostate, Exception, Ref<TrustedHTML>, Ref<TrustedScript>, Ref<TrustedScriptURL>> processValueWithDefaultPolicy(ScriptExecutionContext& scriptExecutionContext, TrustedType expectedType, const String& input, const String& sink)
 {
-    RefPtr<TrustedTypePolicy> protectedPolicy;
+    TrustedTypePolicy* defaultPolicy = nullptr;
```

Comment:
I'm not sure I understand why raw pointers are needed here. Since we eventually call into arbitrary JS via `getPolicyValue` I think we want to protect this object via `RefPtr`.

Reply:
I don't know the spec all that well but my thinking is it's possible to overwrite the default policy via JS so this object could get deleted out from under us without a RefPtr protecting it.

Reply:
It fixes a crash, but I think if I use a locker like on the policy along with a separate change somewhere else it might do the trick. I'll take a look

Reply:
That property is read-only to JS. It can be written once and then never destroyed (until page reload).

Reply:
I think we want to be using RefPtr here regardless.

Reply:
Have swapped this back to using RefPtr, now have a separate non RefPtr version which is used in the opaque root function.

The memory issue is also addressed by removing the trusted types supplement early in the worker destruction process. Thanks @fujii for finding that fix.